src/lib/y2partitioner/sequences/add_partition.rb

```diff
@@ -94,31 +99,55 @@ def size
       end
 
       def role
-        log.info "TODO: Partition ROLE dialog"
-        :next
+        Dialogs::PartitionRole.run(disk.name, @options)
       end
+
       skip_stack :role
 
       def format_mount
         ptable = disk.partition_table
-        name = next_free_primary_partition_name(@disk_name, ptable)
+        name = next_free_partition_name(@disk_name, ptable, @ptemplate.type)
         partition = ptable.create_partition(name, @ptemplate.region, @ptemplate.type)
-        Dialogs::FormatAndMount.new(partition).run
+
+        FormatMount::Base.new(partition, @options).apply_options!
+
+        :finish
+      end
+
+      def format_options
```

Comment:
IMHO the sequence step that displays a "Format and Mount" dialog should also be called `format_mount`. This is confusing, because there are also separate Format Options and Mount Options dialogs. The newly added last step (with `FormatMount::Base`) should be called `commit` or something similar.